rocksdb_admin/admin_handler.cpp

```diff
@@ -981,7 +981,8 @@ void AdminHandler::async_tm_backupDBToS3(
   common::Timer timer(kS3BackupMs);
   LOG(INFO) << "S3 Backup " << request->db_name << " to " << request->s3_backup_dir;
   auto ts = common::timeutil::GetCurrentTimestamp();
-  auto local_path = folly::stringPrintf("%s%s%d/", getS3TmpPath(), request->db_name.c_str(), ts);
+  const std::string s3_tmp_path = getS3TmpPath();
```

Comment:
What if we let`getS3TmpPath` return `const char*` instead? Would that be less error-prone?

Reply:
There are other consumers for getS3TmpPath.

I think a better solution is to avoid folly::stringPrintf in favor of more modern functions.

Reply:
If you dont mind I will do this in a follow up change

Reply:
np, follow up is good